modules/php/States/ResolveFlippedTrait.php

```diff
@@ -0,0 +1,37 @@
+<?php
+namespace BANG\States;
+use BANG\Core\Stack;
+use BANG\Helpers\Utils;
+use BANG\Managers\Cards;
+use BANG\Managers\Players;
+use BANG\Core\Notifications;
+
+trait ResolveFlippedTrait
+{
+  /*
+   * stResolveFlipped: called during the beginning each player turn if Dynamite or Jail required resolving
+   */
+  public function stResolveFlipped()
+  {
+    $atom = Stack::top();
+    $player = Players::get($atom['pId']);
+    $args = ['player' => $player];
+    $cards = Cards::getInLocation(LOCATION_FLIPPED);
+    if ($cards->count() == 1) {
+      $card = $cards->first();
+      $player->discardCard($card, true); // Discard a flipped card
+      $player->discardCard($atom['src'], true); // Discard Jail itself
+
+      if ($card->getCopyColor() == 'H') {
```

Comment:
I would rather have this part inside the card itself, with a method call resolveFlipped or whatever.
Because this state will be triggered by any flip, not only for Jail or Dynamite, so it's better to let the card handle what should happens after a flip.

Reply:
That makes perfect sense, yeah